### PR TITLE
Update checklist log oriented backlog immutable storage , H02.04, G03.11, F03.21

### PR DESCRIPTION
**Pull request by hgotar1**
Add 3 entries in the JSON check lists

H02.04
Establish a process for using code to implement quick fixes. Always register quick fixes in your team's backlog so each fix can be reworked at a later point, and you can limit technical debt.
https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/considerations/automation
G03.11
Use an Azure Event Grid-based solution for log-oriented, real-time alerts
https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/landing-zone/design-area/security
F03.21
Export logs to Azure Storage if your log retention requirements exceed seven years. Use immutable storage with a write-once, read-many policy to make data non-erasable and non-modifiable for a user-specified interval.https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/landing-zone/design-area/management-platform

### CODE REVIEW

**Comment by erjosito** [comment]
Hey @hgotar1 thanks for this! Please do let us know when this PR is ready for review

**Review by mbilalamjad — APPROVED** [review]
Reviewed and approved, added changes for D02.01 and B01.01

**Review by brsteph — APPROVED** [review]

File: checklists/alz_checklist.en.json [hunk 1]
```diff
@@ -2019,6 +2019,16 @@
             "training": "https://azure.github.io/azure-monitor-baseline-alerts/patterns/alz/",
             "link": "https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/landing-zone/design-area/management-monitor"
         },
+        {
+            "category": "Management",
+            "subcategory": "Monitoring",
+            "text": "Export logs to Azure Storage if your log retention requirements exceed seven years. Use immutable storage with a write-once, read-many policy to make data non-erasable and non-modifiable for a user-specified interval.",
```
Comment (brsteph) [inline checklists/alz_checklist.en.json:2025]: @mbilalamjad - This seems to be a duplicate with  "G03.02", should we look to consolidate?

**Review by brsteph — CHANGES_REQUESTED** [review]
Making a change note on the duplicate of "G03.02"

File: checklists/alz_checklist.en.json (hunk 1 above)
Comment (mbilalamjad) [reply checklists/alz_checklist.en.json:2025]: Good catch let me remove this one and lets keep G03.02 for now

**Review by brsteph — APPROVED** [review]

**Pull request merged**